ariwete/pkg/config/config.go

```diff
@@ -207,7 +211,55 @@ func (c *Config) FindSetupFiles(paths []string) (*map[string]CISetup, error) {
 				return nil, err
 			}
 		}
+		err := c.ValidateCISetup(setupFile, setup)
+		if err != nil {
+			return nil, err
+		}
 		setups[path] = setup
 	}
 	return &setups, nil
 }
+
+func (c *Config) ValidateCISetup(path string, setup CISetup) error {
+	var sb strings.Builder
+	for field := range setup {
+		value, exists := c.CISetupDefaults[field]
+		if strings.HasPrefix(field, "_") {
+			// This is a comment field, no need to validate.
+			continue
+		}
+
+		// Check if the field exists.
+		if !exists {
+			sb.WriteString(fmt.Sprintf("%v: '%v' does not exist.\n", path, field))
```

Comment:
Reading this error message, I was confused about "existence", since it does exist in the package specific config file we're looking at. Maybe we could word it in contrast to the central config defaults? Although "default" also doesn't completely describe that file anymore. Perhaps something like: "%v: '%v' unexpected key found.\n"   or something along those lines?